libs/auth/src/dipped_smart_card_auth.ts

```diff
@@ -12,27 +12,40 @@ import {
  * inserted and removed from the card reader for the user to be authenticated
  */
 export interface DippedSmartCardAuthApi {
-  getAuthStatus: () => DippedSmartCardAuth.AuthStatus;
+  getAuthStatus: (
+    machineState: DippedSmartCardAuthMachineState
+  ) => Promise<DippedSmartCardAuth.AuthStatus>;
 
-  checkPin: (input: { pin: string }) => void;
-  logOut: () => void;
+  checkPin: (
+    machineState: DippedSmartCardAuthMachineState,
+    input: { pin: string }
+  ) => Promise<void>;
+  logOut: (machineState: DippedSmartCardAuthMachineState) => Promise<void>;
 
-  programCard: (input: {
-    userRole:
-      | SystemAdministratorUser['role']
-      | ElectionManagerUser['role']
-      | PollWorkerUser['role'];
-  }) => Promise<Result<{ pin?: string }, Error>>;
-  unprogramCard: () => Promise<Result<void, Error>>;
-
-  setElectionDefinition: (electionDefinition: ElectionDefinition) => void;
-  clearElectionDefinition: () => void;
+  programCard: (
+    machineState: DippedSmartCardAuthMachineState,
+    input: {
+      userRole:
+        | SystemAdministratorUser['role']
+        | ElectionManagerUser['role']
+        | PollWorkerUser['role'];
+    }
+  ) => Promise<Result<{ pin?: string }, Error>>;
+  unprogramCard: (
+    machineState: DippedSmartCardAuthMachineState
+  ) => Promise<Result<void, Error>>;
 }
 
 /**
  * Configuration parameters for a dipped smart card auth instance
  */
 export interface DippedSmartCardAuthConfig {
   allowElectionManagersToAccessUnconfiguredMachines?: boolean;
+}
+
+/**
+ * Machine state that the consumer is responsible for providing
+ */
+export interface DippedSmartCardAuthMachineState {
   electionDefinition?: ElectionDefinition;
 }
```

Comment:
Since we're really just checking the election hash from the machine against the hash from the card, should we just pass in the election hash instead of this abstracted machine state? Might be a little easier to grok. Or is there a reason we might want the full election def or want to add other fields to the machine state?

I know card programming is one place where we use the `electionData` string as well, but you could argue that it's more clear to have that be its own argument passed into `programCard` to indicate that we're programming the card with that data (vs all the other methods that are just doing a hash comparison to validate the current card)